spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -369,15 +370,53 @@ public void alterNamespace(String[] namespace, NamespaceChange... changes) throw
   }
 
   @Override
+  // Spark assumes that catalogs CASCADE by default. So we have to eagerly
+  // attempt to drop namespaces and tables, but the CASCADE keyword is still
+  // required to actually drop tables and namespaces as Spark will error out
+  // if any of the recursive deletes are non-empty and the user didn't specify
+  // cascades in their query.
```

Comment:
I don't think that this is a good place for a multi-line comment. Between `@Override` and the method seems odd. I think I would place it closer to the actual behavior. If we have a namespace catalog, then attempt to recursively delete.

Reply:
Ok. Will move that.